### PR TITLE
Load accounts with index page

### PR DESCRIPTION
**Pull request by ChristophWurst**
* load list of accounts with index page (hidden input, recommended at http://backbonejs.org/#Collection-fetch). This reduces the time needed to load the initial data on page load.
* hide list of accounts/folders until they have finished loading and an error occurred. This also allows users to finally delete/edit accounts that are (temporarily) failing.

### CODE REVIEW

File: js/controller/accountcontroller.js [hunk 1]
```diff
@@ -33,23 +33,37 @@ define(function(require) {
...
+
 		$.when(fetchingAccounts).done(function(accounts) {
 			if (accounts.length === 0) {
 				defer.resolve(accounts);
 				Radio.navigation.trigger('setup');
+
+				Radio.ui.trigger('sidebar:accounts');
 			} else {
 				var loadingAccounts = accounts.map(function(account) {
+					require('state').accounts.add(account);
```
Comment (ChristophWurst) [inline js/controller/accountcontroller.js]: - [ ] TODO: the account might already be in the collection, hence we can remove that method call

**Comment by ChristophWurst** [comment]
@jancborchardt @owncloud/mail please review!

**Comment by irgendwie** [comment]
(First account is the dead account)

If one mail account fails, an unspecific message appears on top: 'Error while loading the selected account.' and both accounts appear to be dead:

deadaccounts

But after waiting another few seconds (more than 5 seconds for me) the functional account "unfolds". But the now appearing functional account is not working when trying to change folders or doing anything:

failingaccount

After removing the failing account, everything works as expected.

**Comment by ChristophWurst** [comment]
> (First account is the dead account)

Known problem https://github.com/owncloud/mail/issues/1475 and unrelated to this PR :wink:

**Comment by irgendwie** [comment]
Ok, then the PR works as expected :+1:

**Pull request merged**